File: app/src/main/java/com/example/client/screens/profile/fragment/ProfileFragment.java
package com.example.client.screens.profile.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import com.bumptech.glide.Glide;
import com.example.client.BuildConfig;
import com.example.client.R;
import com.example.client.app.Constants;
import com.example.client.dialog.PrimaryDialog;
import com.example.client.models.event.Event;
import com.example.client.models.profile.ProfileModel;
import com.example.client.screens.login.activity.LoginEmailActivity;
import com.example.client.screens.message.activity.MessageActivity;
import com.example.client.screens.profile.manager_info.ManagerInfoActivity;
import com.example.client.screens.profile.present.ProfilePresent;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

public class ProfileFragment extends Fragment implements  IProfileView,View.OnClickListener {
    private TextView tvVersion;
    private TextView updateInfo,fullname,email;
    private ImageView avatar;
    private LinearLayout logout,subjects,contact;
    private PrimaryDialog dialog;
    private SwipeRefreshLayout refreshLayout;
    private ProfilePresent pPresent;
    private ProfileModel user;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_profile,null);
        tvVersion = view.findViewById(R.id.tvVersion);
        pPresent = new ProfilePresent(this);
        tvVersion.append(BuildConfig.VERSION_NAME);
        updateInfo = view.findViewById(R.id.updateInfo);
        fullname = view.findViewById(R.id.fullname);
        email = view.findViewById(R.id.email);
        updateInfo.setOnClickListener(this);
        avatar = view.findViewById(R.id.avatar);
        logout = view.findViewById(R.id.logout);
        subjects = view.findViewById(R.id.subjects);
        contact = view.findViewById(R.id.contact);
        refreshLayout = view.findViewById(R.id.container);
        refreshLayout.setOnRefreshListener(()-> refreshLayout.setRefreshing(false));
        dialog = new PrimaryDialog();
        dialog.getInstance(getContext());
        subjects.setOnClickListener(this);
        logout.setOnClickListener(this);
        contact.setOnClickListener(this);

        pPresent.getUserFromRes();
        return view;
    }

    @Override
    public void onStart() {
        super.onStart();
        if (!EventBus.getDefault().isRegistered(this))
            EventBus.getDefault().register(this);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEvent(Event event) {
        switch (event.getKey()) {
            case Constants.EventKey.UPDATE_PROFILE_AVATAR:
            case Constants.EventKey.UPDATE_PROFILE_INFO:
                pPresent.getUserFromRes();
                break;
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        EventBus.getDefault().unregister(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.updateInfo:
                startActivity(new Intent(getContext(), ManagerInfoActivity.class));
                break;
            case R.id.logout:
                dialog.setDescription("Bạn có muốn đăng xuất?");
                dialog.setOKListener(()-> pPresent.onLogout());
                dialog.setCancelListener(()->{});
                dialog.show();
                break;
            case R.id.subjects:

                break;
            case R.id.contact:
                Intent i = new Intent(getContext(), MessageActivity.class);
                i.putExtra("user",user.getId());
                startActivity(i);
                break;
        }
    }

    @Override
    public void logout(boolean isSuccess) {
        if(isSuccess) {
            startActivity(new Intent(getActivity(), LoginEmailActivity.class));
        }
        else {
            Toast.makeText(getActivity(),"Đã xảy ra lỗi!",Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void showUserInfo(ProfileModel profile) {
        this.user = profile;
        fullname.setText(profile.getFullname());
        email.setText(profile.getEmail());
        Glide.with(this)
                .asBitmap()
                .placeholder(R.drawable.avatar_default)
                .load(profile.getAvatar())
                .into(avatar);
    }
}

File: app/src/main/java/com/example/client/screens/profile/fragment/ProfileFragment.kt
package com.example.client.screens.profile.fragment

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.example.client.R
import com.example.client.base.BaseFragmentMVP
import com.example.client.dialog.PrimaryDialog
import com.example.client.models.profile.ProfileModel
import com.example.client.screens.login.activity.LoginEmailActivity
import com.example.client.screens.message.activity.MessageActivity
import com.example.client.screens.order.history.activity.OrderHistoryActivity
import com.example.client.screens.profile.manager_info.activity.ManagerProfileActivity
import com.example.client.screens.profile.present.IProfilePresent
import com.example.client.screens.profile.present.ProfilePresent
import com.example.client.screens.rating.community.CommunityActivity
import kotlinx.android.synthetic.main.fragment_profile.*

class ProfileFragment : BaseFragmentMVP<IProfilePresent>(), IProfileView, View.OnClickListener {

    companion object {
        @JvmStatic
        fun newInstance() = ProfileFragment()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return LayoutInflater.from(context).inflate(R.layout.fragment_profile, null)
    }

    override fun bindEvent() {
        tv_update_info.setOnClickListener(this)
        lnl_logout.setOnClickListener(this)
        lnl_order_history.setOnClickListener(this)
        lnl_community.setOnClickListener(this)
        lnl_contact.setOnClickListener(this)
        swipe_refresh.setOnRefreshListener { hideLoading() }
    }

    override fun bindComponent() {
        presenter.bindData()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_update_info -> startActivity(ManagerProfileActivity.newInstance(requireActivity()))
            R.id.lnl_logout -> {
                PrimaryDialog({
                    presenter.onLogout()
                }, {})
                        .setDescription(getString(R.string.dialog_question_logout))
                        .show(childFragmentManager)
            }
            R.id.lnl_order_history -> startActivity(OrderHistoryActivity.newInstance(requireActivity()))
            R.id.lnl_contact -> presenter.navigateToContact()
            R.id.lnl_community -> startActivity(CommunityActivity.newInstance(requireActivity()))
        }
    }

    override fun logout() {
        requireActivity().finish()
        startActivity(LoginEmailActivity.newInstance(requireActivity()))
    }

    override fun showUserInfo(profile: ProfileModel) {
        tv_full_name.text = profile.fullname
        tv_email.text = profile.email
        Glide.with(this)
                .asBitmap()
                .placeholder(R.drawable.avatar_default)
                .load(profile.avatar)
                .into(view_icon)
    }

    override fun showErrorMessage(errMessage: Int) {
    }

    override fun toContactScreen(id: Int) {
        val i = Intent(context, MessageActivity::class.java)
        i.putExtra("user", id)
        startActivity(i)
    }

    override fun showLoading() {
        swipe_refresh.isRefreshing = true
    }

    override fun hideLoading() {
        swipe_refresh.run {
            if (isRefreshing) isRefreshing = false
        }
    }

    override fun onBackPress() {
    }

    override val presenter: IProfilePresent
        get() = ProfilePresent(this)

}